### PR TITLE
chore(query): Reopen keep-alive for MySQL & rollback socket2 to 0.4

### LINKED ISSUE #10641: bug: Reopen tcp keep-alive for MySQL
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

1.0.21

### What's Wrong?

image
image
Connection cannot be released because there is no keep-alive.

We need to rollback socket2 to 0.4, and reopen keep-alive.

See https://github.com/rust-lang/socket2/issues/402, and wait https://github.com/tokio-rs/tokio/pull/5514.

### How to Reproduce?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by yufan022**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #10641

### CODE REVIEW

**Comment by sundy-li** [comment]
cc @TCeason It's strange that the test result contain grant to system.one.

```
[SQL] SHOW GRANTS FOR 'test-grant'@'localhost'
[Diff] (-expected|+actual)
    GRANT ALL ON 'default'.'default'.* TO 'test-grant'@'localhost'
    GRANT SELECT ON 'default'.'db01'.* TO 'test-grant'@'localhost'
-   GRANT SELECT ON 'default'.'db01'.'tb1' TO 'test-grant'@'localhost'
-   GRANT SELECT ON 'default'.'system'.'one' TO 'test-grant'@'localhost'
+   GRANT SELECT ON 'default'.'db01'.'tb1' TO 'test-grant'@'localhost'
at tests/sqllogictests/suites/base/05_ddl/05_0006_ddl_grant_privilege:49
```

**Comment by BohuTANG** [comment]
@mergify update

**Review by dantengsky — APPROVED** [review]
LGTM

File: src/query/service/Cargo.toml [hunk 1]
```diff
@@ -135,7 +135,7 @@ scopeguard = "1.1.0"
 serde = { workspace = true }
 serde_json = { workspace = true }
 serde_urlencoded = "0.7.1"
-socket2 = "0.5.1"
+socket2 = "0.4.7"
```
Comment (Xuanwo) [inline src/query/service/Cargo.toml:138]: Can you add a link here so that we can upgrade this after tokio released?

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**